### PR TITLE
Update splunk.search to use splunklib.results.ResultsReader

### PR DESCRIPTION
**Pull request by cbrenton**
### Status
- bugfix
### Description

Update splunk.search to use splunklib.results.ResultsReader instead of returning raw data.
### Checklist (tick everything that applies)
- [ ] Metadata: pack.yaml, icon, structure (required for new packs)
- [x] Version bump (required for changed packs)
- [x] Code linting (required, can be done after the PR checks)
- [ ] [Tests](https://docs.stackstorm.com/development/pack_testing.html) (not required but really recommended)

### CODE REVIEW

File: packs/splunk/actions/search.py [hunk 1]
```diff
@@ -20,7 +21,16 @@ def __init__(self, config):
...
-
-        return result.read()
+        result = self.service.jobs.oneshot(query, params={"output_mode": "json"})
+        reader = results.ResultsReader(result)
+        search_results = []
+
+        for item in reader:
+            if isinstance(item, results.Message):
+                # Diagnostic messages may be returned in the results
+                print '%s: %s' % (item.type, item.message)
```
Comment (lakshmi-kannan) [inline packs/splunk/actions/search.py]: Is it intentional that you are using print? You might want to use the logger from the action.

**Comment by Kami** [comment]
Thanks.

Is it possible to add some unit tests or would this require too much mocking and would defeat the purpose?

**Comment by cbrenton** [comment]
I am working on unit tests and will push a new version soon.

**Comment by Kami** [comment]
Great, keep us posted and let us know when it's ready for a review :)

**Comment by LindsayHill** [comment]
Updated here: https://github.com/StackStorm-Exchange/stackstorm-splunk/pull/2

**Pull request closed without merging**